sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -239,6 +239,7 @@ primaryExpression
     | ident ('.' ident)*                                                             #dereference
     | primaryExpression CAST_OPERATOR dataType                                       #doubleColonCast
     | timestamp=primaryExpression AT TIME ZONE zone=primaryExpression                #atTimezone
+    | 'ARRAY'? '[]'                                                                  #emptyArray
```

Comment:
This was necessary to work around some ambiguity we've as we support the `ARRAY[]` constructor also without `ARRAY` prefix (which PostgreSQL doesn't)